src/main/java/org/broadinstitute/hellbender/tools/walkers/GnarlyGenotyper.java

```diff
@@ -0,0 +1,292 @@
+
+    /**
+     * This option can only be activated if intervals are specified.
+     */
+    @Advanced
+    @Argument(fullName = GenotypeGVCFs.ONLY_OUTPUT_CALLS_STARTING_IN_INTERVALS_FULL_NAME,
+            doc="Restrict variant output to sites that start within provided intervals",
+            optional=true)
+    private boolean onlyOutputCallsStartingInIntervals = false;
+
+    @Argument(fullName = GenomicsDBImport.MERGE_INPUT_INTERVALS_LONG_NAME,
+            shortName = GenomicsDBImport.MERGE_INPUT_INTERVALS_LONG_NAME,
+            doc = "Boolean flag to read in all data in between intervals.  Improves performance reading from GenomicsDB " +
+                    "using large lists of +intervals, as in exome sequencing, especially if GVCF data only exists for " +
+                    "specified intervals.")
+    private boolean mergeInputIntervals = false;
+
+    @Hidden
+    @Argument(fullName = "strip-allele-specific-annotations", shortName = "strip-as", doc = "Remove raw AS values and don't calculate finalized values")
+    private boolean stripASAnnotations = false;
+
+
+    /**
+     * The rsIDs from this file are used to populate the ID column of the output.  Also, the DB INFO flag will be set
+     * when appropriate. Note that dbSNP is not used in any way for the genotyping calculations themselves.
+     */
+    @ArgumentCollection
+    private final DbsnpArgumentCollection dbsnp = new DbsnpArgumentCollection();
+
+    private VariantContextWriter vcfWriter;
+    private VariantContextWriter annotationDatabaseWriter = null;
+    private GnarlyGenotyperEngine genotyperEngine;
+    private final RMSMappingQuality mqCalculator = RMSMappingQuality.getInstance();
+    private final Set<Class<? extends InfoFieldAnnotation>> allAlleleSpecificAnnotations = new HashSet<>();
+
+    /** these are used when {@link #onlyOutputCallsStartingInIntervals) is true */
+    private List<SimpleInterval> intervals;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    /**
+     * Get the largest interval per contig that contains the intervals specified on the command line.
+     * @param getIntervals intervals to be transformed
+     * @param sequenceDictionary used to validate intervals
+     * @return a list of one interval per contig spanning the input intervals after processing and validation
+     */
+    @Override
+    protected List<SimpleInterval> transformTraversalIntervals(final List<SimpleInterval> getIntervals, final SAMSequenceDictionary sequenceDictionary) {
+        if (mergeInputIntervals) {
+            return IntervalUtils.getSpanningIntervals(getIntervals, sequenceDictionary);
+        } else {
+            return getIntervals;
+        }
+    }
+
+    @Override
+    protected GenomicsDBOptions getGenomicsDBOptions() {
+        if (genomicsDBOptions == null) {
+            genomicsDBOptions = new GenomicsDBOptions(referenceArguments.getReferencePath(), true, PIPELINE_MAX_ALT_COUNT);
```

Comment:
Declare the middle `true` as a named constant at the top of the class, for clarity's sake.